pkg/controllers/externalsecret/externalsecret_controller_secret.go

```diff
@@ -113,7 +113,7 @@ func (r *Reconciler) handleGenerateSecrets(ctx context.Context, namespace string
 	if err != nil {
 		return nil, err
 	}
-	secretMap, err := gen.Generate(ctx, genDef, r.Client, namespace)
+	secretMap, err := gen.Generate(ctx, genDef, r.Client, namespace, r.ControllerClass)
```

Comment:
I think the generator doesn't need to be aware of the `controllerClass`. We can make the decision (controllerClass matches or not) before we call `gen.Generate()`.

All information is in `genDef` which we can get out. Consider this: 

```go
// a type that implements just enough for a controllerClass.
// a generator _can_ implement this, it is optional though
type controllerClassResource {
  Spec  struct{
    ControllerClass string `json:"controllerClass"`
  } `json:"spec"`
}

var cc controllerClassResource
json.Umarshal(genDef, &cc)

if cc.Spec.ControllerClass != "" && cc.Spec.ControllerClass != r.controllerClass {
  return errors.New("invalid controller class")
}
```

That way we implement it centrally here instead of having it implemented per-generator. WDYT?

Reply:
Hi @moolen,

I think this is a good idea to centralize the implementation. Will follow the pattern, Thanks for the idea.